runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -11651,6 +11994,585 @@ static TR::Register *inlineIntrinsicIndexOf(TR::Node *node, bool isLatin1, TR::C
    return result;
    }
 
+/*
+ * Arraycopy evaluator needs a version of inlineArrayCopy that can be used inside internal control flow. For this version of inlineArrayCopy, registers must
+ * be allocated outside of this function so the dependency at the end of the control flow knows about them.
+ */
+static void inlineArrayCopy_ICF(TR::Node *node, int64_t byteLen, TR::Register *src, TR::Register *dst, TR::CodeGenerator *cg, TR::Register *cndReg,
+                                TR::Register *tmp1Reg, TR::Register *tmp2Reg, TR::Register *tmp3Reg, TR::Register *tmp4Reg,
+                                TR::Register *fp1Reg, TR::Register *fp2Reg, TR::Register *fp3Reg, TR::Register *fp4Reg)
+   {
+   if (byteLen == 0)
+      return;
+
+   TR::Register *regs[4] = {tmp1Reg, tmp2Reg, tmp3Reg, tmp4Reg};
+   TR::Register *fpRegs[4] = {fp1Reg, fp2Reg, fp3Reg, fp4Reg};
+   int32_t groups, residual;
+
+   static bool disableLEArrayCopyInline = (feGetEnv("TR_disableLEArrayCopyInline") != NULL);
+   TR_Processor processor = TR::Compiler->target.cpu.id();
+   bool supportsLEArrayCopyInline = (processor >= TR_PPCp8) && !disableLEArrayCopyInline && TR::Compiler->target.cpu.isLittleEndian() && TR::Compiler->target.cpu.hasFPU() && TR::Compiler->target.is64Bit();
+
+   if (TR::Compiler->target.is64Bit())
+      {
+      groups = byteLen >> 5;
+      residual = byteLen & 0x0000001F;
+      }
+   else
+      {
+      groups = byteLen >> 4;
+      residual = byteLen & 0x0000000F;
+      }
+
+   int32_t regIx = 0, ix = 0, fpRegIx = 0;
+   int32_t memRefSize;
+   TR::InstOpCode::Mnemonic load, store;
+   TR::Compilation* comp = cg->comp();
+
+   /* Some machines have issues with 64bit loads/stores in 32bit mode, ie. sicily, do not check for is64BitProcessor() */
```

Comment:
@IBMJimmyk: Do you have a pointer to `sicily`? I can't find it anywhere.

Reply:
I don't think it's an external machine available on the openj9 jenkins instance; The comment should've probably described the machine configuration rather than the name.

Reply:
Is it possible for you to provide enough information for someone to be able to improve the comment?

I presume this is something related to https://en.wikipedia.org/wiki/IBM_POWER_microprocessors -- but there are a lot of items in that list.

(I found this as part of #5722 -- specifically the `s` family...)

I'm happy to update the comment, but as it isn't available to the openj9 instance, I won't be able to figure out what to write ...

(Alternatively, if you're likely to fix the comment, I can leave this alone.)

Reply:
@zl-wang do you recall our discussion here?